packages/go/src/symbols/named-type.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  Children,
+  Namekey,
+  OutputSpace,
+  Refkey,
+  track,
+  TrackOpTypes,
+  trigger,
+  TriggerOpTypes,
+} from "@alloy-js/core";
+import { GoSymbol, GoSymbolOptions } from "./go.js";
+
+// represents a symbol from a .go file. Struct, interface, etc.
+export type NamedTypeTypeKind =
+  | "import"
+  | "function"
+  | "var"
+  | "type"
+  | "struct"
+  | "struct-member"
+  | "interface"
+  | "interface-member"
+  | "package";
+export type NamedTypeSymbolKind = "named-type" | "package";
+
+/**
+ * Information for a Go generic type parameter.
+ */
+export interface TypeParameterProps {
+  /**
+   * The name of the type parameter.
+   */
+  name: string;
+
+  /**
+   * The parameter constraint
+   */
+  constraint: Children;
+
+  /**
+   * A refkey for this type parameter.
+   */
+  refkey?: Refkey;
+}
+
+export interface NamedTypeSymbolOptions extends GoSymbolOptions {
+  typeParameters?: TypeParameterProps[];
+}
+
+/**
+ * A symbol for a named type in Go such as a struct, interface, and so forth.
+ */
+export class NamedTypeSymbol extends GoSymbol {
+  public readonly symbolKind: NamedTypeSymbolKind = "named-type";
+  public static readonly memberSpaces = ["members"];
+
+  constructor(
+    name: string | Namekey,
+    spaces: OutputSpace[] | OutputSpace | undefined,
+    kind: NamedTypeTypeKind,
+    options?: NamedTypeSymbolOptions,
+  ) {
+    super(name, spaces, options);
+    this.#typeKind = kind;
+    this.#typeParameters = options?.typeParameters;
+  }
+
+  #typeKind: NamedTypeTypeKind;
+  get typeKind() {
+    track(this, TrackOpTypes.GET, "typeKind");
+    return this.#typeKind;
+  }
+  set typeKind(value: NamedTypeTypeKind) {
+    const old = this.#typeKind;
+    if (old === value) {
+      return;
+    }
+    this.#typeKind = value;
+    trigger(this, TriggerOpTypes.SET, "typeKind", value, old);
+  }
+
+  #typeParameters?: TypeParameterProps[];
+  get typeParameters() {
+    track(this, TrackOpTypes.GET, "typeParameters");
+    return this.#typeParameters;
+  }
+  set typeParameters(value: TypeParameterProps[] | undefined) {
+    const old = this.#typeParameters;
+    if (old === value) {
+      return;
+    }
+    this.#typeParameters = value;
+    trigger(this, TriggerOpTypes.SET, "typeParameters", value, old);
+  }
+
+  copy() {
+    const options = this.getGoCopyOptions();
+    const copy = new NamedTypeSymbol(
+      this.name,
+      undefined,
+      this.#typeKind,
+      options,
+    );
+    this.initializeGoCopy(copy);
+    return copy;
+  }
+
+  #pointerSymbol?: GoSymbol;
```

Comment:
Noted above but it seems problematic to have a single pointer symbol for a given type. Each type can have multiple pointers in different scopes no?